plugins/snowflake/dbt/include/snowflake/macros/adapters.sql

```diff
@@ -1,14 +1,37 @@
 {% macro snowflake__create_table_as(temporary, relation, sql) -%}
   {%- set transient = config.get('transient', default=true) -%}
+  {%- set cluster_by_keys = config.get('cluster_by', default=none) -%}
+  {%- set enable_automatic_clustering = config.get('automatic_clustering', default=false) -%}
+  {%- if cluster_by_keys is not none and cluster_by_keys is string -%}
+    {%- set cluster_by_keys = [cluster_by_keys] -%}
+  {%- endif -%}
+  {%- if cluster_by_keys is not none -%}
+    {%- set cluster_by_string = cluster_by_keys|join(", ")-%}
+  {% else %}
+    {%- set cluster_by_string = none -%}
+  {%- endif -%}
+
+      create or replace {% if temporary -%}
+        temporary
+      {%- elif transient -%}
+        transient
+      {%- endif %} table {{ relation }}
+      as (
+        {%- if cluster_by_string is not none -%}
+          select * from(
+            {{ sql }}
+            ) order by ({{ cluster_by_string }})
+        {%- else -%}
+          {{ sql }}
+        {%- endif %}
+      );
+    {% if cluster_by_string is not none -%}
+      alter table {{relation}} cluster by ({{cluster_by_string}});
+    {%- endif -%}
+    {% if enable_automatic_clustering and cluster_by_string is not none  -%}
```

Comment:
```suggestion
    {% if enable_automatic_clustering and cluster_by_string is not none and not temporary  -%}
```

Same here - i don't think we'll want to resume reclustering on a temp table!